java-manual/modules/ROOT/pages/transactions.adoc

```diff
@@ -0,0 +1,487 @@
+
+For these more advanced use-cases, the driver provides functions to take full control over the transaction lifecycle. These are called _managed transactions_, and you can think of them as a way of unwrapping the flow of `executableQuery()` and being able to specify its desired behavior in more places.
+
+
+== Create a session
+
+Before running a transaction, you need to obtain a _session_.
+Sessions act as concrete query channels between the driver and the server, and ensure <<causal consistency>> is enforced.
+
+Sessions are created with the method link:https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/Driver.html#session()[`Driver.session()`].
+Use the optional argument to alter the session's configuration, among which for example the xref:_database_selection[target database].
+For further configuration parameters, see xref:_session_configuration[Session configuration].
+
+[source, java]
+----
+// import org.neo4j.driver.SessionConfig
+
+try (var session = driver.session(SessionConfig.builder().withDatabase("neo4j").build())) {
+    ...
+----
+
+Session creation is a lightweight operation, so sessions can be created and destroyed without significant cost.
+Always xref:_close_sessions[close sessions] when you are done with them.
+
+*Sessions are _not_ thread safe*: you can share the main `Driver` object across threads, but make sure each thread creates its own sessions.
+
+
+[#managed-transactions]
+== Run a managed transaction
+
+A transaction can contain any number of queries.
+As Neo4j is <<ACID>> compliant, *queries within a transaction will either be executed as a whole or not at all*: you cannot get a part of the transaction succeeding and another failing.
+Use transactions to group together related queries which work together to achieve a single _logical_ database operation.
+
+A managed transaction is created with the methods `Session.executeRead()` and `Session.executeWrite()`, depending on whether you want to retrieve data from the database or alter it.
+Both methods take a _transaction function_ callback, which is responsible for actually carrying out the queries and processing the result.
+
+.Retrieve people whose name starts with `Al`.
+[source, java]
+----
+// import java.util.Map
+// import org.neo4j.driver.SessionConfig
+
+try (var session = driver.session(SessionConfig.builder().withDatabase("neo4j").build())) {  // <1>
+    var people = session.executeRead(tx -> {  // <2>
+        var result = tx.run("""
```

Comment:
This example shows only one statement either. There is no difference towards the executableQuery variant.

Reply:
This is very true. I didn't want to make it into too a complicated example, since it already has 4 callouts, but I get your point that it doesn't really display a _transaction_. At this stage, the intention is mostly to teach you how to use transaction functions, and then the next example will show real transactions. Do you think this is a poor intention? Would it make it better to have a code comment after the first query saying smth like "run more queries"?

Reply:
My compromise on this is
- leave the example with one query as is, but add a comment saying more queries would get executed in the same tx. I'd rather have the example a bit more stupid but where the reader can entirely focus on the driver API, as meaningful transaction examples are much longer (there's one just after)
- in the quickstart page, put the long hard transaction example.